baselines/jft/active_learning.py

```diff
@@ -489,93 +633,167 @@ def main(config, output_dir):
       num_epochs=1,  # Don't repeat
   )
 
+  current_opt_repl = flax_utils.replicate(opt_cpu)
+  pool_ids, _, _, pool_masks = get_ids_logits_masks(
+      model=model,
+      opt_repl=current_opt_repl,
+      ds=pool_train_ds,
+  )
+
+  rng, initial_uniform_rng = jax.random.split(rng)
+  pool_scores = get_uniform_scores(pool_masks, initial_uniform_rng)
+
+  acquisition_batch_ids, _ = select_acquisition_batch_indices(
+        acquisition_batch_size=config.get("acquisition_batch_size", 10),
+        scores=pool_scores,
+        ids=pool_ids,
+        ignored_ids=set(),
+  )
+
+  # NOTE: if we could `enumerate` before `filter` in `create_dataset` of CLU
+  # then this dataset creation could be simplified.
+  # https://github.com/google/CommonLoopUtils/blob/main/clu/deterministic_data.py#L340
+  # CLU is explicitly not accepting outside contributions at the moment.
+  train_subset_data_builder = tfds.builder(
+      "cifar10_subset", subset_ids={
+          "train": set(acquisition_batch_ids),
+          "test": None
+      })
+  train_subset_data_builder.download_and_prepare()
+
   test_accuracies = []
-  training_size = []
+  training_sizes = []
 
-  rng, rng_loop = jax.random.split(rng, 2)
+  rng, rng_loop = jax.random.split(rng)
   rngs_loop = flax_utils.replicate(rng_loop)
 
+  # NOTE: it's VITAL train_ds_rng is used for all train_ds creations
+  rng, train_ds_rng = jax.random.split(rng)
+
   current_train_ds_length = len(train_subset_data_builder.subset_ids["train"])
-  while current_train_ds_length < config.get("max_labels", 150):
+  while current_train_ds_length < config.get("max_labels", 250):
     print(f"Training set size: {current_train_ds_length}")
 
-    if current_train_ds_length > 0:
-      rng, train_ds_rng = jax.random.split(rng)
-      train_ds_rng = jax.random.fold_in(train_ds_rng, jax.process_index())
+    current_opt_repl = flax_utils.replicate(opt_cpu)
+
+    # Repeat dataset to have oversampled epochs and bootstrap more batches
+    number_of_batches = current_train_ds_length / config.batch_size
+    num_repeats = math.ceil(config.total_steps / number_of_batches)
+    print(f"Repeating dataset {num_repeats} times")
+
+    repeated_train_ds = input_utils.get_data(
+        dataset=train_subset_data_builder,
+        split=config.train_split,
+        rng=train_ds_rng,
+        process_batch_size=local_batch_size,
+        preprocess_fn=preprocess_spec.parse(
+            spec=config.pp_train, available_ops=preprocess_utils.all_ops()),
+        shuffle_buffer_size=config.shuffle_buffer_size,
+        prefetch_size=config.get("prefetch_to_host", 2),
+        # TODO: double check if below actually leads to bootstrap sampling
+        num_epochs=num_repeats,
+    )
+
+    train_ds = input_utils.get_data(
```

Comment:
I think it would be helpful to rename this to something like `train_eval_ds` since it's intended to be used for eval only (e.g., isn't shuffled and is using the eval preprocessing (`pp_eval`)).